### PR TITLE
New delegate page

### PR DESCRIPTION
**Pull request by Levia**
A delegate can be added to a respondent by selecting an existing delegate or creating a new one.

### CODE REVIEW

File: app/controllers/user_delegates_controller.rb [hunk 1]
```diff
@@ -24,29 +25,38 @@ def create
...
         if @user_delegate.save
-          UserMailer.delegate_registration(delegate, password, "http://#{request.host}/").deliver
+          UserMailer.delegate_request(@user_delegate).deliver
           success = true
+        elsif UserDelegate.find_by_user_id_and_delegate_id(@user_delegate.user_id, @user_delegate.delegate_id)
+          flash[:error] = t('user_delegates.email_already_a_delegate')
+          success = true
+        end
+      else
+        password = SecureRandom.hex.first(8)
```
Comment (spencerldixon) [inline app/controllers/user_delegates_controller.rb]: You don't need the `.first` here, you can do `SecureRandom.hex(8)`

Comment (Levia) [reply app/controllers/user_delegates_controller.rb]: SecureRandom.hex(8) and SecureRandom.hex.first(8) are two different things: https://ruby-doc.org/stdlib-2.2.0/libdoc/securerandom/rdoc/SecureRandom.html#method-c-hex

SecureRandom.hex(8) will return a random string of length 16, while doing first(8) I'm getting just the first 8 characters

Comment (spencerldixon) [reply app/controllers/user_delegates_controller.rb]: Ah ok, in that case then I think `.first(8)` is a bit more intuitive than doing `.hex(4)`, let's leave it as is

Comment (Levia) [reply app/controllers/user_delegates_controller.rb]: Yep, that should make sense. Updated!

**Pull request merged**